### ISSUE TITLE
Multi-Objective Hyper-parameter optimization

### ISSUE THREAD

**Issue by oxinabox** [baggepinnen/Hyperopt.jl#5]
This is another rather blue-sky thing, but I thought I would note it down.

Often when I want to do hyper-parameter optimization I have multiple metrics I would like to optimize for.

I would like some kind of multiobjective hyperparameter optimization
that can search for and fine the optimal frontier.

Then after that is done I can look at the optimal frontier and know what my options are,
then choise one of the points to actually use.

I think it is much easier for humans to pick a point on a phaseplot of metrics, if they know their options,
than it is for them to to describe up front a single compound metric that captures their desires

**Comment by baggepinnen** [comment]
This could potentially be added with rather small effort, at least to some of the optimizers. Most optimizers store the entire history and it would just be a matter of storing more than one metric visualizing it properly.

**Comment by oxinabox** [comment]
For random search yes,
but for anything smarter you need a multiobjective way of deciding where to explore next.

**Comment by norci** [comment]
is this code useful?

https://github.com/optuna/optuna/blob/master/optuna/multi_objective/samplers/_motpe.py

> Multi-objective sampler using the MOTPE algorithm.
>    This sampler is a multiobjective version of :class:`~optuna.samplers.TPESampler`.
